sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -69,6 +69,19 @@ internal class StyleObserver(
    */
   override fun onStyleLoaded(eventData: StyleLoadedEventData) {
     val style = loadedStyle ?: throw MapboxMapException("Style is not initialized on onStyleLoaded callback!")
+
+    // sprites listener is not null - it means style does not contain them thus StyleDataType.SPRITE
+    // was not fired, explicitly call it to bind all the necessary style extensions upstream
+    if (styleDataSpritesLoadedListener != null) {
+      styleDataSpritesLoadedListener?.onStyleLoaded(style)
+      styleDataSpritesLoadedListener = null
```

Comment:
we should explicitly set to null before `styleDataSpritesLoadedListener?.onStyleLoaded(style)` otherwise we might have an issue if `styleDataSpritesLoadedListener?.onStyleLoaded(style)` will trigger any update in the same callchain